src/spatialmath/quaternion.rs

```diff
@@ -0,0 +1,371 @@
+use float_cmp::{ApproxEq, F64Margin, approx_eq};
+
+use super::vector3::Vector3;
+
+
+/// A Rust implementation of Quaternion, we use this instead of existing packages
+/// because we want a C-safe representational structure. These quaternions use
+/// the Real-I-J-K standard, so those using JPL or other standards should take
+/// care to convert before initializing a Quaternion struct from the library
+#[repr(C)]
+#[derive(Copy, Clone, Debug)]
+pub struct Quaternion {
+    pub real: f64,
+    pub i: f64,
+    pub j: f64,
+    pub k: f64,
+}
+
+impl Quaternion {
+    /// Initializes a quaternion from a real component and an
+    /// imaginary 3-vector
+    pub fn new_from_vector(real: f64, imag: Vector3) -> Self {
```

Comment:
This is hopefully my own naiveté, but is this a common way to initialize a quaternion? Why would 3 of the values already be in a vector?

Reply:
sometimes when you've converted it from an axis angle (with vector and theta) you scale the axis angle vector by sin(theta/2), so you might want to keep it as a vector then? I have a follow-up PR that uses is this but I'm not attached to the convenience so I'm willing to remove